data_adapter_oemof/adapters.py

```diff
@@ -59,146 +40,118 @@ def get_default_parameters(cls, struct: dict, mapper: Mapper) -> dict:
         return defaults
 
 
-def facade_adapter(cls):
-    r"""
-    Sets type of Busses to str
-    Sets 'build_solph_components' to False in __init__.
-    """
-    # set type of bus to str
-    for field in dataclasses.fields(cls):
-        if field.type == Bus:
-            field.type = str
-
-    original_init = cls.__init__
-
-    # do not build solph components
-    def new_init(self, *args, **kwargs):
-        custom_attributes = {}
-        original_fields = tuple(field.name for field in dataclasses.fields(cls))
-        for key in tuple(kwargs.keys()):
-            if key not in original_fields:
-                custom_attributes[key] = kwargs.pop(key)
-
-        original_init(self, *args, build_solph_components=False, **kwargs)
-
-        for key, value in custom_attributes.items():
-            setattr(self, key, value)
-
-    cls.__init__ = new_init
-
-    return cls
-
-
-@facade_adapter
-class DispatchableAdapter(facades.Dispatchable, Adapter):
+class DispatchableAdapter(Adapter):
     """
     Dispatchable Adapter
     """
 
     type = "dispatchable"
+    facade = facades.Dispatchable
```

Comment:
I think we have had this Version with facade as DataAdapter attribute before but I do not remember why we changed that.